### DISCUSSION TITLE
Combining any() and between() for an ARRAY column

### DISCUSSION THREAD

**Discussion by Dukisef012** [sqlalchemy/sqlalchemy#9389] [Usage Questions]
I'm trying to filter all rows which have an array field, whose at least one element is in the range of two values. So far I had no luck in writing something that works.

**Comment by CaselIT** [comment]
Hi,

could you share your current effort and what sql are you trying to get?
I'm guess postgresql, but again just guessing

**Reply by Dukisef012** [reply]
Hey @CaselIT, something like this:
`select(MyModel).where(MyModel.array.any(between(min, max)))`

**Reply by CaselIT** [reply]
What sql should that compile to?

**Reply by Dukisef012** [reply]
I'm not really sure, never written something like this, and not an experienced SQL writer, guess something like this:
`SELECT *
FROM my_table
WHERE EXISTS (
    SELECT 1
    FROM unnest(my_array) AS x
    WHERE x BETWEEN min_value AND max_value
);`

**Reply by CaselIT** [reply]
I don't think there is anything automatic, no. cc @zzzeek do you remember if array has anything like that?

It should be possible to generate though. Something like
```py
unnest = sa.func.unnest(MyModel.array).table_valued('x').render_derived('w')
exist = sa.exists().select_from(unnest).where(unnest.c.x.between(1,2))
```
then you can place exist in the where clause

**Reply by zzzeek** [reply]
i dont recall anything about unnest() coming out automatically
